### PR TITLE
fix generated handle() to remove 'Content-Length' header. fixes #83

### LINKED ISSUE #83: ensure correct Content-Length header on outbound requests
Many of the outbound integrations set the `Content-Length` header using the string length of the body being sent in the request (e.g., `headers['Content-Length'] = body.length`). But for bodies with any multi-byte characters, that's wrong, and causes the request sent to be truncated. `Buffer.byteLength()` is the right way to set this.

However, for all the integrations that use `request()` & `response()` (vs. their own `handle()` function), this module generates a `handle()` function, which is what's used to perform the request. This should be fixed to correct the `Content-Length` header, rendering the problem moot at the integration level. 

(In fact, `request()`/`response()` integrations should not set this header at all, as that value will be disregarded anyway.)

### PR DESCRIPTION
**Pull request by cgrayson**

### CODE REVIEW

File: src/index.coffee [hunk 1]
```diff
@@ -145,6 +145,9 @@ generateHandle = (outbound) ->
     catch err
       return callback(err)
 
+    # remove Content-Length header and let the request library set it (correctly, on byte length, not character length)
+    delete outboundReq.headers['Content-Length'] if outboundReq.headers?
+
```
Comment (alexkwolfe) [inline src/index.coffee]: I wonder if setting this to `null` would be sufficient? We've read in the v8 JS engine there are performance penalties for deleting properties this way.

Comment (cgrayson) [inline src/index.coffee]: Well, here's the thing if we null it: 

screen shot 2016-02-19 at 4 19 19 pm

Comment (alexkwolfe) [inline src/index.coffee]: What about setting it to `undefined`. I bet that will work. IIRC, MongoDB will not persist a property with an `undefined` value.

Comment (cgrayson) [inline src/index.coffee]: Duh, should have thought of that. Perfect.

**Pull request merged**